ruff.toml

```diff
@@ -0,0 +1,73 @@
+# Exclude a variety of commonly ignored directories.
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".ipynb_checkpoints",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pyenv",
+    ".pytest_cache",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    ".vscode",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "site-packages",
+    "venv",
+    "env",
+]
+
+# Same as Black.
+line-length = 88
+indent-width = 4
+
+# Assume Python 3.11
+target-version = "py311"
+
+[lint]
+select = [
+    # https://docs.astral.sh/ruff/rules/#error-e
+    "E",
+    # https://docs.astral.sh/ruff/rules/#pyflakes-f
+    "F",
+]
+ignore = [
+    # Whitespace before ':' (conflicts with Black)
+    "E203",
+    # Two spaces before inline comment
+    "E261",
+    # Line too long
+    "E501",
+    # Do not assign a lambda expression
+    "E731",
+    # Name may be undefined from '*' import
+    "F405",
+    # Blank line contains whitespace
+    "W293",
+    # Trailing whitespace
+    "W291",
```

Comment:
This seems like a useful check. Why ignore it? Have to admint I'm biased on this one. I've configured my editor to automatically delete trailing whitespace on save, so whenever I come across a file that contains trailing whitespace somewhere this is a bit of a nuisance.